gateleen-queue/src/main/java/org/swisspush/gateleen/queue/expiry/ExpiryCheckHandler.java

```diff
@@ -78,14 +66,8 @@ public static void updateServerTimestampHeader(MultiMap headers) {
 
             headers.set(SERVER_TIMESTAMP_HEADER, nowAsISO);
         } else {
-            String updatedTimestamp = localizeTimestamp(serverTimestamp);
-
-            if (!updatedTimestamp.equals(serverTimestamp)) {
-                log.debug("Updating" + SERVER_TIMESTAMP_HEADER + " value from " + serverTimestamp + "  to " + updatedTimestamp);
-
-                headers.remove(SERVER_TIMESTAMP_HEADER);
-                headers.set(SERVER_TIMESTAMP_HEADER, updatedTimestamp);
-            }
+            headers.remove(SERVER_TIMESTAMP_HEADER);
+            headers.set(SERVER_TIMESTAMP_HEADER, printDateTime(DateTime.now()));
```

Comment:
Is it by intent to remove that condition?

Reply:
I was actually uncertain here. Hope that @ljucam can say something here.

Reply:
Before approving we should be clear about the removal of this condition.

Reply:
Honestly, I’m not really sure. Before this change, we updated the timestamp only then, when it differed to the current localized timestamp. Now we directly set it to the current timestamp. So, we actively always change the server timestamp, even if it would be right. Before that, we just converted it. Therefore, I’m not sure, what could change. 


```
private static String localizeTimestamp(String timestamp) {
        String localizedTimestamp = timestamp;
        if (localizedTimestamp != null && localizedTimestamp.toUpperCase().endsWith("Z")) {
            try {
                DateTime dt = isoDateTimeParser.parseDateTime(localizedTimestamp);
                localizedTimestamp = dfISO8601.print(dt);
            } catch (IllegalArgumentException e) {
                log.warn("Could not parse: " + localizedTimestamp);
            }
        }

        return localizedTimestamp;
    }
```

Reply:
If we cannot explain what we need the equals() check for then I suggest that we remove it and do exactly what it says in the method description:

"If no valid value is found, the actual timestamp is set."

Reply:
I am OK with this